### DISCUSSION TITLE
Collab Text / TipTap Editor

### DISCUSSION THREAD

**Discussion by MentalGear** [aspen-cloud/triplit#95] [General]
Currently researching ways to connect triplit with collab CRDT text. I also looked at several different collab editors recently, and TipTap seems the most ergonomic and also elegantly abstracted API on top of Prosemirror.

Now I was thinking about how to best connect triplit with [TipTaps/Y.js collab guide](https://tiptap.dev/docs/editor/collaboration/install) and wondering about the best approach? Would the Y.js State just be stored in a Triplit field, or is there more needed for this?

I'd appreciate any insights ! :)

Since triplit seems also to be working on a crdt collab text solution

_Other References:_
Besides Y.js, there's also `list-positions` CRDT solution. Its author provide some very welcome demo for [triplit/quell text demo](https://github.com/mweidner037/list-demos/tree/master).

**Comment by MentalGear** [comment]
As per the roadmap it seems like @matlin and team are working on an own collab text type. This [issue with some interesting thoughts](https://github.com/mweidner037/list-demos/issues/7#issuecomment-2075894659) by the `list-positions` author might be of possible interest.

**Comment by MentalGear** [comment]
Thank you for your input in the linked issue.

It seems like the only way to currently have live collaborative rich text editing with tree data structures - which is require by a good chunk of best in class editors (TipTap / ProseMirror-based) - is to have a separate y-prosemirror instance running that syncs to triplit server-side. So quite a bit hacky.

May we expect the upcoming native triplit **collab text feature** (Q2) to be a tree data structure (which would be in line with the overall store) so an easy integration with ProseMirror-based editors can be done? :)

**Reply by matlin** [reply]
I'm not sure exactly what strategy we'll ultimately go with because this is a greenfield area with a lot of active research but having broad editor support would be a high priority.

**Reply by MentalGear** [reply]
Collab/version control is indeed a very fluid field. I just received [new research](https://buttondown.email/geoffreylitt/archive/towards-universal-version-control-with-patchwork/) from the excellent research group of ink & switch describing their new patchwork collaborative/version control system. You might be interested as well. Maybe there's a way their system could be used directly in triplit's stack.
